### PR TITLE
Implemented endpoint to fetch all channels in Tribler

### PR DESCRIPTION
**Pull request by devos50**
Also did some small refactoring in `channels_endpoint.py`.

### CODE REVIEW

**Comment by devos50** [comment]
@lfdversluis ready for review

File: Tribler/Core/Modules/restapi/channels_endpoint.py [hunk 1]
```diff
@@ -52,15 +55,33 @@ class ChannelsSubscribedEndpoint(BaseChannelsEndpoint):
...
-        for channel in subscribed_channels_db:
-            results_json.append(self.convert_db_channel_to_json(channel))
-
+        results_json = [self.convert_db_channel_to_json(channel) for channel in subscribed_channels_db]
         return json.dumps({"subscribed": results_json})
+
+
+class ChannelsAllEndpoint(BaseChannelsEndpoint):
+    """
+    A GET request to this endpoint returns all channels in Tribler.
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: All channels that Tribler has received from peers, you have no guarantee that you know all channels yet. @qstokkink is doing experiments with this stuff :D

File: Tribler/Test/Core/Modules/RestApi/test_channels_endpoints.py [hunk 2]
```diff
@@ -49,3 +50,29 @@ def test_get_subscribed_channels_one_subscription(self):
                                         expected_json[u'subscribed'][0][u'description'])
         self.vote_for_channel(cid, expected_json[u'subscribed'][0][u'modified'])
         return self.do_request('channels/subscribed', expected_code=200, expected_json=expected_json)
+
+    @deferred(timeout=10)
+    def test_get_all_channels_no_channels(self):
+        """
+        Testing whether the API returns no channels when fetching all channels and there are no created channels
```
Comment (lfdversluis) [inline Tribler/Test/Core/Modules/RestApi/test_channels_endpoints.py]: no channels created AND in the DB i guess?

File: Tribler/Test/Core/Modules/RestApi/test_channels_endpoints.py [hunk 3]
```diff
@@ -49,3 +50,29 @@ def test_get_subscribed_channels_one_subscription(self):
...
+        channels_json = json.loads(channels)
+        self.assertEqual(len(channels_json['channels']), 10)
+        for i in range(len(channels_json['channels'])):
+            self.assertEqual(channels_json['channels'][i]['name'], 'Test channel %d' % i)
+            self.assertEqual(channels_json['channels'][i]['description'], 'Test description %d' % i)
+
+    @deferred(timeout=10)
+    def test_get_all_channels_no_channels(self):
+        """
+        Testing whether the API returns channels when fetching all channels
```
Comment (lfdversluis) [inline Tribler/Test/Core/Modules/RestApi/test_channels_endpoints.py]: returns all channels known in the DB*

**Comment by lfdversluis** [comment]
pylint `function-redefined`, Please fix

**Comment by devos50** [comment]
Implemented feedback. @whirm ready for review.

File: Tribler/Core/Modules/restapi/channels_endpoint.py (hunk 1 above)
Comment (whirm) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: Yeah, that should be called ChannelsDiscoveredEndpoint or similar.

Comment (whirm) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: Ok Known channels

Comment (devos50) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: @whirm should this also be reflected in the URL (so make it `/channels/known` or `/channels/discovered`)?

**Comment by whirm** [comment]
Other than the naming/description, looks fine to me.

**Comment by devos50** [comment]
@whirm done, see my last three fixup commits.

**Comment by whirm** [comment]
Looks good, squash

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
```
test_ffsearch (Tribler.Test.test_remote_search.TestMixedRemoteSearch) ... Unhandled Error
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Utilities/twisted_thread.py", line 55, in _reactor_runner
    reactor.run(installSignalHandlers=False)
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1194, in run
    self.mainLoop()
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 1203, in mainLoop
    self.runUntilCurrent()
--- <exception caught here> ---
  File "/usr/lib/python2.7/dist-packages/twisted/internet/base.py", line 798, in runUntilCurrent
    f(*a, **kw)
  File "/home/jenkins/workspace/GH_Tribler_PR_tests_linux/tribler/Tribler/Core/Modules/tracker_manager.py", line 98, in update_tracker_info
    tracker_info = self._tracker_dict[tracker_url]
exceptions.TypeError: 'NoneType' object has no attribute '__getitem__'
```

I've seen this error before and iirc I already created an issue for it.

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@whirm done, ready for merge

**Pull request merged**